cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/ProgramRuntimeProviderLoader.java

```diff
@@ -87,7 +68,7 @@ public ProgramRuntimeProviderLoader(CConfiguration cConf) {
   @Nullable
   public ProgramRuntimeProvider get(ProgramType programType) {
     try {
-      ProgramRuntimeProvider provider = programRunnerProviderCache.get(programType);
+      ProgramRuntimeProvider provider = programRuntimeExtensionLoader.getExtension(programType);
       if (provider != NOT_SUPPORTED_PROVIDER) {
```

Comment:
Will the extension loader ever returns `NOT_SUPPORTED_PROVIDER`?? I think the `getExtension` call will throw exception.